src/index.js

```diff
@@ -18,201 +15,227 @@ const logger = {
-        request(signInOptions, (err, res, body) => {
-            let errType = "";
-            if (err) {
-                errType = errors.VENDOR_DOWN;
+    // Login layer
+    async login(loginToken, requiredFields) {
+        const signInOptions = {
+            method: "POST",
+            ecdhCurve: "auto",
+            url: `https://www.materiel.net/form/submit_login`,
+            form: {
+                Email: requiredFields.login,
+                Password: requiredFields.password,
+                __RequestVerificationToken: loginToken
             }
-            else {
-                try {
-                    body = JSON.parse(body);
-                    if (!body || !body.authenticationSuccess || !body.user) {
-                        if (typeof(body.loginForm) === "string" && body.loginForm.includes('window.renderCaptcha()')) {
-                            errType = 'USER_ACTION_NEEDED.CAPTCHA';
-                        }
-                        else {
-                            errType = errors.LOGIN_FAILED;
+        };
+
+        return new Promise((resolve, reject) => {
+            logger.info("Signing in");
+            this.request(signInOptions, (err, res, body) => {
+                let errType = "";
+                if (err) {
+                    errType = errors.VENDOR_DOWN;
+                }
+                else {
+                    try {
+                        body = JSON.parse(body);
+                        if (!body || !body.authenticationSuccess || !body.user) {
+                            if (typeof(body.loginForm) === "string" && body.loginForm.includes('window.renderCaptcha()')) {
+                                errType = 'USER_ACTION_NEEDED.CAPTCHA';
+                            }
+                            else {
+                                errType = errors.LOGIN_FAILED;
+                            }
                         }
                     }
+                    catch (e) {
+                        console.log(e)
+                        logger.error("Cannot parse response");
+                        errType = errors.LOGIN_FAILED;
+                    }
                 }
-                catch (e) {
-                    logger.error("Cannot parse response");
-                    errType = errors.LOGIN_FAILED;
-                }
-            }
 
-            if (errType) {
-                logger.error("Signin failed");
-                return reject(new Error(errType));
-            }
+                if (errType) {
+                    logger.error("Signin failed");
+                    return reject(new Error(errType));
+                }
 
-            let cookie = `ID=${body.user.Id}&KEY=${body.user.AuthenticationCode}`;
-            j.setCookie(`Customer=${cookie}`, "https://secure.materiel.net");
+                let cookie = `ID=${body.user.Id}&KEY=${body.user.AuthenticationCode}`;
+                this._jar.setCookie(`Customer=${cookie}`, "https://secure.materiel.net");
```

Comment:
Reference to class cookie jar

Reply:
Pre-existing but could you reuse `baseUrl` please?